homeassistant/components/homewizard/sensor.py

```diff
@@ -414,16 +411,111 @@ class HomeWizardSensorEntityDescription(SensorEntityDescription):
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_name="Heat meter",
+    ),
+    ExternalDevice.DeviceType.WARM_WATER_METER: HomeWizardExternalSensorEntityDescription(
+        key="warm_water_meter",
+        translation_key="total_liter_m3",
+        suggested_device_class=SensorDeviceClass.WATER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_name="Warm water meter",
+    ),
+    ExternalDevice.DeviceType.WATER_METER: HomeWizardExternalSensorEntityDescription(
+        key="water_meter",
+        translation_key="total_liter_m3",
+        suggested_device_class=SensorDeviceClass.WATER,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_name="Water meter",
+    ),
+    ExternalDevice.DeviceType.INLET_HEAT_METER: HomeWizardExternalSensorEntityDescription(
+        key="inlet_heat_meter",
+        translation_key="total_energy_gj",
+        suggested_device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_name="Inlet heat meter",
+    ),
+}
+
+
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Initialize sensors."""
     coordinator: HWEnergyDeviceUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
-    async_add_entities(
+
+    # Migrate original gas meter sensor to ExternalDevice
+    ent_reg = er.async_get(hass)
+
+    if (
+        entity_id := ent_reg.async_get_entity_id(
+            Platform.SENSOR, DOMAIN, f"{entry.unique_id}_total_gas_m3"
+        )
+    ) and coordinator.data.data.gas_unique_id is not None:
+        ent_reg.async_update_entity(
+            entity_id,
+            new_unique_id=f"{DOMAIN}_{coordinator.data.data.gas_unique_id}",
+        )
+
+    # Migrate original gas meter identifier sensor to ExternalDevice
+    if (
+        entity_id := ent_reg.async_get_entity_id(
+            Platform.SENSOR, DOMAIN, f"{entry.unique_id}_gas_unique_id"
+        )
+    ) and coordinator.data.data.gas_unique_id is not None:
+        ent_reg.async_update_entity(
+            entity_id,
+            new_unique_id=f"{DOMAIN}_{coordinator.data.data.gas_unique_id}_meter_identifier",
+        )
+
+    # Initialize default sensors
+    entities: list = [
         HomeWizardSensorEntity(coordinator, description)
         for description in SENSORS
         if description.has_fn(coordinator.data.data)
-    )
+    ]
+
+    # Initialize external devices
+    if coordinator.data.data.external_devices is not None:
+        for unique_id, device in coordinator.data.data.external_devices.items():
+            if description := EXTERNAL_SENSORS.get(device.meter_type):
+                device_info = DeviceInfo(
+                    identifiers={(DOMAIN, unique_id)},
+                    name=description.device_name,
+                    manufacturer="HomeWizard",
+                    model=coordinator.data.device.product_type,
```

Comment:
```suggestion
                    model=coordinator.data.device.product_type,
                    serial_number=unique_id,
```

Reply:
We can remove the sensor and that will be enough IMO. But as it removes the sensor, which is migrated from the original configuration, we should mark it as breaking I think.

Reply:
@emontnemery Is the breaking-change label still required or will it be picked up by the PR description?